### PR TITLE
#181135878

### PR DESCRIPTION
**Pull request by SalShai**
## Feature description
- This Function Fetches the Salary Components from a given Salary Structure.
- It further calculates the amount of each component if the component amount is based on a formula.  In that case, it checks if the base amount exists. If notify it throws an error.
- If the amount of components is not based on a formula, the item just fetches the given amount.
- This action is taken before saving the file.

## Solution description
- Populate Salary Structure Components from a given Salary Structure.
- Calculate the Amount based on Formula (if any).
- Calculate Indemnity Amount and Leave Allocation Amount if chosen.

## Output screenshots (optional)

https://user-images.githubusercontent.com/29017559/152936323-3ee6b8c2-74c6-44c3-88e6-38ca9a3e1d9c.mov

## Areas affected and ensured
Salary Structure Assignment Doctype Modified.

## Is there any existing behavior change of other features due to this code change?
Yes,
- Modification in Salary Structure Assignment doctype to fetch and calculate Indemnity Amount and Leave Allocation Amount.

## Was this feature tested on all the browsers?
  - [x] Chrome
  - [x] Safari

### CODE REVIEW

**Comment by pjamsheer** [comment]
@SalShai What is the use of one_fm/public/js/doctype_js/salary_structure_assignment.js

File: one_fm/one_fm/hr_utils.py [hunk 1]
```diff
@@ -23,10 +24,15 @@ def indemnity_allocation_builder(employee_list):
...
     indemnity_allcn.employee = employee.name
-    indemnity_allcn.from_date = employee.date_of_joining
-    indemnity_allcn.new_indemnity_allocated = 30/365
-    indemnity_allcn.total_indemnity_allocated = 30/365
-    indemnity_allcn.submit()
+    salary = get_reference_salary(employee)
+    # indemnity_allcn.from_date = employee.date_of_joining
+    # indemnity_allcn.new_indemnity_allocated = 30/365
+    # indemnity_allcn.total_indemnity_allocated = 30/365
+    # indemnity_allcn.submit()
```
Comment (pjamsheer) [inline one_fm/one_fm/hr_utils.py]: How the Indemnity will be allocated, if we commented this lines.

File: one_fm/one_fm/hr_utils.py [hunk 2]
```diff
@@ -23,10 +24,15 @@ def indemnity_allocation_builder(employee_list):
 def create_indemnity_allocation(employee):
     indemnity_allcn = frappe.new_doc('Indemnity Allocation')
     indemnity_allcn.employee = employee.name
-    indemnity_allcn.from_date = employee.date_of_joining
-    indemnity_allcn.new_indemnity_allocated = 30/365
-    indemnity_allcn.total_indemnity_allocated = 30/365
-    indemnity_allcn.submit()
+    salary = get_reference_salary(employee)
```
Comment (pjamsheer) [inline one_fm/one_fm/hr_utils.py]: Here in the indemnity allocation, We are just adding the factor of to the indemnity allocation. The Salary and amount details we need in the report for indemnity Calculation !

File: one_fm/one_fm/hr_utils.py [hunk 3]
```diff
@@ -8,13 +8,14 @@
...
-        select emp.name, emp.date_of_joining, emp.indemnity_policy
+        select emp.name, emp.date_of_joining
         from `tabEmployee` emp
         left join `tabIndemnity Allocation` ia on emp.name = ia.employee and ia.docstatus = 1 and emp.status = 'Active'
         where ia.employee is NULL
     """
     employee_list = frappe.db.sql(query, as_dict=True)
-    frappe.enqueue(indemnity_allocation_builder, timeout=600, employee_list=employee_list)
+    print(employee_list)
+    # frappe.enqueue(indemnity_allocation_builder, timeout=600, employee_list=employee_list)
```
Comment (pjamsheer) [inline one_fm/one_fm/hr_utils.py]: How the Indemnity will be allocated, if we commented this lines.

File: one_fm/one_fm/hr_utils.py (hunk 1 above)
Comment (SalShai) [reply one_fm/one_fm/hr_utils.py]: > @SalShai What is the use of one_fm/public/js/doctype_js/salary_structure_assignment.js

It's currently empty and has no script.

File: one_fm/public/js/doctype_js/salary_structure_assignment.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+frappe.ui.form.on('Salary Structure Assignment', {
+    // refresh: function(frm) {
+
+	  // }
+});
```
Comment (pjamsheer) [inline one_fm/public/js/doctype_js/salary_structure_assignment.js]: Do we need this file

File: one_fm/hooks.py [hunk 5]
```diff
@@ -76,6 +76,7 @@
 	"Employee Incentive": "public/js/doctype_js/employee_incentive.js",
 	"Employee": "public/js/doctype_js/employee.js",
 	"Salary Slip": "public/js/doctype_js/salary_slip.js",
+	"Salary Structure Assignment": "public/js/doctype_js/salary_structure_assignment.js",
```
Comment (pjamsheer) [inline one_fm/hooks.py]: js file is empty !!

**Review by mymi14s — APPROVED** [review]

**Pull request merged**